cora/util/cubicspline.pyx

```diff
@@ -110,15 +109,13 @@ cdef class Interpolater(object):
         return self.value(x)
 
     @cython.boundscheck(False)
-    def value_array(self, double[::1]x):
+    def value_array(self, x):
 
-        cdef double[:] xr
-        cdef double[:] rr
 
         cdef Py_ssize_t i, nr
 
-        xr = np.asarray(x).ravel()
-        rr = np.empty_like(xr)
+        cdef double[::1] xr = np.ascontiguousarray(x.ravel())
+        cdef double[:] rr = np.empty_like(xr)
```

Comment:
This probably is a little more optimal as

```suggestion
        r = np.empty_like(x)
        cdef double[::1] rr = r.ravel()        
```
and then just `return r` at the end.